src/tasksToDo.js

```diff
@@ -0,0 +1,116 @@
+export default class TasksToDo {
+  constructor() {
+    this.arrayTasks = [];
+  }
+
+  newTask = (desc, completed, index) => {
+    const task = {
+      desc,
+      completed,
+      index,
+    };
+    this.arrayTasks.push(task);
+    localStorage.setItem('localTasks', JSON.stringify(this.arrayTasks));
+    window.location.reload();
+  };
+
+  displayTasks = () => {
+    const tasks = document.querySelectorAll('.item');
+    tasks.forEach((item, index) => {
+      item.addEventListener('click', () => {
+        const itemContent = document.querySelector(`.item${index}`);
+        const trashImage = document.querySelector(`.trash-image${index}`);
+        const moreImage = document.querySelector(`.hamburger-image${index}`);
+        const selected = document.querySelector('.selected');
+        const trashImageActive = document.querySelector('.trash-image-show');
+        const moreImageHide = document.querySelector('.more-image-hide');
+        if (selected != null) {
+          selected.classList.remove('selected');
+          trashImageActive.classList.remove('trash-image-show');
+          moreImageHide.classList.remove('more-image-hide');
+        }
+        itemContent.classList.add('selected');
+        trashImage.classList.add('trash-image-show');
+        moreImage.classList.add('more-image-hide');
+      });
+    });
+  }
+
+  adjustDescription = () => {
+    let lenght = 0;
+    this.arrayTasks.forEach((item) => {
+      // eslint-disable-next-line no-self-assign
+      item.desc = item.desc;
+      item.index = lenght;
+      // eslint-disable-next-line no-self-assign
+      item.completed = item.completed;
+      lenght += 1;
+    });
+  }
+
+  deleteTask = (btn) => {
+    btn.forEach((btn, index) => {
+      btn.addEventListener('click', () => {
+        this.arrayTasks = this.arrayTasks.filter((item, i) => i !== index);
+        // let length = 0;
+        // this.arrayTasks.forEach((item) => {
+        //   // eslint-disable-next-line no-self-assign
+        //   item.desc = item.desc;
+        //   item.index = length;
+        //   // eslint-disable-next-line no-self-assign
+        //   item.completed = item.completed;
+        //   length += 1;
+        // });
+        this.adjustDescription();
+        localStorage.setItem('localTasks', JSON.stringify(this.arrayTasks));
+        window.location.reload();
+      });
+    });
+  }
+
+  updateTasksList = (iDescription) => {
+    iDescription.forEach((desc, index) => {
```

Comment:
- [ ] Kindly fix this issue, when I tick the box I got this error.
<img width="1402" alt="Screenshot 2023-05-06 at 13 07 04" src="https://user-images.githubusercontent.com/81444338/236644823-b39aa729-be2c-47b2-907f-798a9e99eb85.png">

Reply:
@Kossi-stack thanks, kindly can you help me with more information about the error? When running on my computer, it doesn't show me any problem when I check the box, please help me with any revision suggestions that I have to do. Thank you.